src/utilities/oas-operation/oas-operation.ts

```diff
@@ -0,0 +1,50 @@
+import { Operation, ParameterObject } from 'swagger-client';
+import { ResponseObject } from 'swagger-client/schema';
+import ExampleGroup, { ExampleGroupFactory } from '../example-group';
+
+class OASOperation {
+  private operationId: string;
+
+  private _schema: Operation;
+
+  private exampleGroups: ExampleGroup[];
+
+  constructor(schema: Operation) {
+    this._schema = schema;
+    this.operationId = schema.operationId;
+    this.exampleGroups = ExampleGroupFactory.buildFromOperation(this);
```

Comment:
I think this one i like keeping a getter method (although i'll convert it to an actual get method) since i imagine we don't get ts warnings from pushing or popping a readonly array.